tutormfe/plugin.py

```diff
@@ -86,6 +74,27 @@ def _add_core_mfe_apps(apps: dict[str, MFE_ATTRS_TYPE]) -> dict[str, MFE_ATTRS_T
     return apps
 
 
+@hooks.Actions.PROJECT_ROOT_READY.add()
+def _modify_mfes(root: str):
+    """
+    Set the GitHub ref for OpenEdX MFEs that are using the *.master branch.
+
+    If the repository of the MFE is not of openedx or the branch is not a
+    *.master branch, we are not setting the ref, hence the build won't benefit
+    from build caches.
+    """
+    config = tutor_config.load_minimal(root)
```

Comment:
This is kind of a hack... I understand why you need to do this, but I really wish we did not have to do that. 

So, let me suggest an alternative solution. In the Dockerfile, write:

```
{% set app_ref = get_app_ref(app, MFE_COMMON_VERSION) %}
{% if app_ref %}
ADD {{ app_ref }} /tmp/gitref-{{ app_name }}
{% endif %}
```

And in plugin.py:

```
def get_app_ref(app, common_version):
   return "" or ...
tutor_hooks.Filters.ENV_TEMPLATE_VARIABLES.add_item(("get_app_ref", get_app_ref))
```